toolkit/tools/imagecustomizerapi/dirconfig.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+)
+
+// DirConfigList is a list of destination files where the source file will be copied to in the final image.
+// This type exists to allow a custom marshaller to be attached to it.
+type DirConfigList []DirConfig
+
+type DirConfig struct {
+	// The path in the target OS that the directory will be copied to.
+	SourcePath string `yaml:"SourcePath"`
+
+	// The path in the target OS that the directory will be copied to.
+	DestinationPath string `yaml:"DestinationPath"`
+
+	// The permissions to set on the directory.
+	Permissions *DirPermissions `yaml:"Permissions"`
+
+	// The permissions to set on the children of the directory.
+	ChildPermissions *DirPermissions `yaml:"ChildPermissions"`
+}
+
+func (l *DirConfigList) IsValid() (err error) {
+	if len(*l) <= 0 {
+		return fmt.Errorf("list is empty")
```

Comment:
is there a case where this would be less than 0?

Reply:
I do not think so, but I chose to keep this format since it is also used in `fileconfig.go`: https://github.com/microsoft/azurelinux/blob/862c25d1ce38634a8a8b20453d8b8d6ec1ac6b42/toolkit/tools/imagecustomizerapi/fileconfig.go#L35